### ISSUE TITLE
Python stubs for static type checking

### ISSUE THREAD

**Issue by altosaar** [apache/incubator-mxnet#8359]
Static type checking helps avoid bugs - would be great to have an mxnet stub here:

https://github.com/python/typeshed

**Comment by KellenSunderland** [comment]
I agree.  I see in the release notes for mxnet 12 RC0 that they're providing IDEs like pycharm with code completion hints.  I wonder if this means someone has already generated stubs?

https://cwiki.apache.org/confluence/display/MXNET/MXNet+0.12.0+Release+Notes

**Comment by eric-haibin-lin** [comment]
ping @szha who worked on IDE integration

**Comment by szha** [comment]
Yes, stubs are already generated, though it's not pyi. Let me know if you need pointers on the code generation.

One potential problem in type checking mxnet API is that some APIs may return more than one types, and the return type depends on the argument, due to legacy design.

**Comment by KellenSunderland** [comment]
Pinging @tdomhan , who was also hoping to see this feature.
